cmake/BLTMacros.cmake

```diff
@@ -979,3 +979,86 @@ macro(blt_combine_static_libraries)
 
     unset( libLinkLine )
 endmacro(blt_combine_static_libraries)
+
+
+
+##------------------------------------------------------------------------------
+## blt_print_target_properties (TARGET <target> )
+##
+## Prints out all properties of the given target.
+##
+## The required target parameteter must either be a valid cmake target 
+## or was registered via blt_register_library.
+##
+## Output is of the form:
+##     [<target> property] <property>: <value>
+## for each property
+##------------------------------------------------------------------------------
+macro(blt_print_target_properties)
+
+    set(options)
+    set(singleValuedArgs TARGET)
+    set(multiValuedArgs)
+
+    ## parse the arguments to the macro
+    cmake_parse_arguments(arg
+         "${options}" "${singleValuedArgs}" "${multiValuedArgs}" ${ARGN})
+
+    ## check for required arguments
+    if(NOT DEFINED arg_TARGET)
+        message(FATAL_ERROR "TARGET is a required parameter for the blt_print_target_properties macro")
+    endif()
+
+    if(TARGET ${arg_TARGET})
```

Comment:
You should error out if not a cmake or blt registered target

Reply:
I think it should be ok here since it is a diagnostic.

On the flip side -- Is there a way to check if something is a registered target?  I guess it has ``BLT_FOO_INCLUDES`` defined?

Reply:
True... I am fine with it not erroring out but can we add a message at the start and/or end that indicates it is printing information.  That way when its empty they know something has gone wrong.

Reply:
I'll add a comment to indicate whether the target is a cmake target, a registered target,  neither or both.

Reply:
Fantastic.

Reply:
Done.

Here is the output for 'not-a-target':
```cmake
-- [blt_print_target_properties] Invalid argument 'not-a-target'. This macro applies only to valid cmake targets or blt_registered targets.
```